src/lib/utils.ts

```diff
@@ -22,7 +22,7 @@ const statusSymbols = {
  * Find the index of a task line in an array of lines by its ID.
  * Supports both emoji format (🆔 abc123) and Dataview format ([[id:: abc123]])
  */
-function findTaskLineByIdOrText(
+export function findTaskLineByIdOrText(
```

Comment:
Meh, these utils require a rewrite. For this PR I think it is the right approach 👍 , but I'm not a fan of exporting a "subfunction" (how it was intended) as interface now in the utils. 

Ideally task entities should know their own location. Keeping this state in sync between memory (Task entities) and the markdown files could be tricky and is already going wrong when tasks are moved between files.